src/Post/domain/entity/Post.ts

```diff
@@ -0,0 +1,39 @@
+import { get } from 'http';
+import { PostInterface } from './PostInterface';
+
+export class Post {
+  private id?: string;
+  public category?: string;
+  public content: string;
+  public userid: string;
+  public userName: string;
+
+  constructor(data: PostInterface) {
+    this.id = data.id;
+    this.category = data.category;
+    this.content = data.content;
+    this.userid = data.userid;
+    this.userName = data.userName;
+  }
+
+  get Id() {
+    if (this.id) return this.id;
+    else return '';
```

Comment:
I think we don't need to return empty string here

Reply:
Thanks Tu. will fix it

Reply:
If I don't return empty string, there was compiler error since my Id is optional data type here.
What should we do to make it better here?

Reply:
I have updated to short version as shown below for now. But the functionality is still the same.

```
 get Id() {
    return this.id ? this.id : '';
  }
```

Reply:
I think Post is an entity so it should have an ID without empty string